### ISSUE TITLE
Uninstallation seems to be done via Package Name

### ISSUE THREAD

**Issue by farosch** [martinet101/WingetUI#288]
It looks like uninstallations are triggered using the Package Name.
This does not work when two installed packages have the same name.
Installations/uninstallations should be triggered using the Package ID.

![wingetui_Bznrfl8GJB](https://user-images.githubusercontent.com/22905106/196195282-cbc9f757-03d4-4e22-a5ae-a8dc87a20385.png)

**Comment by ppvnf** [comment]
I agree this should be implemented for installs, uninstalls and updates as well. Also, I believe it's better to use --id and -e, like suggested in Winget [docs](https://learn.microsoft.com/en-us/windows/package-manager/winget/install):

> The best way to limit the selection to one file is to use the id of the application combined with the exact query option. For example:
> `winget install --id Git.Git -e `

**Comment by fernandog** [comment]
A quick look at the code I found this:

https://github.com/martinet101/WingetUI/blob/main/wingetui/storeEngine.py#L374

If uses ID and "-e" only if `self.useif` is set

```
if(self.store == "winget" or self.store.lower() == "local pc"):
            self.p = subprocess.Popen(self.adminstr + [wingetHelpers.winget, "uninstall", "-e"] + (["--id", self.packageId] if self.useId else ["--name", self.programName]) + wingetHelpers.common_params + self.cmdline_args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, stdin=subprocess.PIPE, shell=True, cwd=sudoLocation, env=os.environ)
```

I found out this code `useId=not("…" in id)`

So, if it founds the "..." in the ID, code uses package name.

Question is: why package would have "..." ? Maybe too long string ?

Maybe @martinet101 can enlighten us

**Comment by martinet101** [comment]
> Question is: why ID would have "..." ? Maybe too long string ?

We should ask this to winget. 

Whenever some package names/ids are _too long_, winget cuts them, ignoring if there's plenty of horixontal space in the console window. When so happens, it uses name.

However, this case shouldn't apply here and winget should be defaulting to ID uninstall

**Comment by fernandog** [comment]
Do you have an example of package name/ID that this issue happens @martinet101 ?

**Comment by fernandog** [comment]
Did you try the "Format-Table -AutoSize" parameter for Powershell / subprocess.Popen ?

https://learn.microsoft.com/en-GB/powershell/module/microsoft.powershell.utility/format-table

**Comment by martinet101** [comment]
I used the `mode` command in cmd, and while it works, winget just ignores it. There's an issue where this was discussed, but it is just winget's fault

**Comment by fernandog** [comment]
winget with Win11 beta (insider) has no issue. If I resize Powershell window, strings go to the next line

![image](https://user-images.githubusercontent.com/2620870/196235651-0a69d728-5717-45f2-895d-3be1f1befbc7.png)

![image](https://user-images.githubusercontent.com/2620870/196235895-06503292-3936-48f3-a4a3-f02614635de6.png)

**Comment by fernandog** [comment]
> which winget version are you using?

PS C:\WINDOWS\system32> winget --version
v1.3.2691

**Comment by martinet101** [comment]
the command `winget search ""` also shows the full names?

**Comment by fernandog** [comment]
Look like it's showing f ull names

![image](https://user-images.githubusercontent.com/2620870/196236688-30de637d-c597-454f-a605-788bed023443.png)

**Comment by fernandog** [comment]
damn, truncated

![image](https://user-images.githubusercontent.com/2620870/196237168-e20e23d3-9cb4-44d5-8a40-ea1ae6b09fb0.png)

**Comment by martinet101** [comment]
That's the thing i was saying. it sometimes does work, it sometimes does not

**Comment by ppvnf** [comment]
I've tested it here on cmd and PShell. If the window is maximized I get the full results, if it's a regular window I get the ...

**Comment by fernandog** [comment]
@ppvnf its not the console window. Its how winget print the results on screen when you run `winget search " "`

**Comment by fernandog** [comment]
Opened an issue there: https://github.com/microsoft/winget-cli/issues/2603

**Comment by martinet101** [comment]
The issue with the uninstall failing has been fixed
